In [1]:
import pandas as pd

In [5]:
df = pd.read_csv("heart.csv")

In [11]:
print("Punto 1.1")
df.head()

Punto 1.1


age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [15]:
print("Punto 1.2: Estadisticas Descriptivas")
df.describe()

Punto 2.2: Estadisticas Descriptivas


age         sex          cp    trestbps        chol         fbs  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean    54.366337    0.683168    0.966997  131.623762  246.264026    0.148515   
std      9.082101    0.466011    1.032052   17.538143   51.830751    0.356198   
min     29.000000    0.000000    0.000000   94.000000  126.000000    0.000000   
25%     47.500000    0.000000    0.000000  120.000000  211.000000    0.000000   
50%     55.000000    1.000000    1.000000  130.000000  240.000000    0.000000   
75%     61.000000    1.000000    2.000000  140.000000  274.500000    0.000000   
max     77.000000    1.000000    3.000000  200.000000  564.000000    1.000000   

          restecg     thalach       exang     oldpeak       slope          ca  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean     0.528053  149.646865    0.326733    1.039604    1.399340    0.729373   
std      0.525860   22.905161    0.469794    1.161075    0.616226    1.022606   
min      0.000000   71.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000  133.500000    0.000000    0.000000    1.000000    0.000000   
50%      1.000000  153.000000    0.000000    0.800000    1.000000    0.000000   
75%      1.000000  166.000000    1.000000    1.600000    2.000000    1.000000   
max      2.000000  202.000000    1.000000    6.200000    2.000000    4.000000   

             thal      target  
count  303.000000  303.000000  
mean     2.313531    0.544554  
std      0.612277    0.498835  
min      0.000000    0.000000  
25%      2.000000    0.000000  
50%      2.000000    1.000000  
75%      3.000000    1.000000  
max      3.000000    1.000000

In [23]:
print("Punto 2.1: Eliminar fila 1")
df.drop(1).head()

Punto 2.1: Eliminar fila 1


age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [35]:
print("Punto 2.2: Eliminar columna 'ca'")
df.drop("ca", axis=1).head()

Punto 2.2: Eliminar columna 'ca'


age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   thal  target  
0     1       1  
1     2       1  
2     2       1  
3     2       1  
4     2       1

In [39]:
print("Punto 2.3: Eliminar columna 'cp'")
df.drop("cp", axis=1).head()

Punto 2.3: Eliminar columna 'cp'


age  sex  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  ca  \
0   63    1       145   233    1        0      150      0      2.3      0   0   
1   37    1       130   250    0        1      187      0      3.5      0   0   
2   41    0       130   204    0        0      172      0      1.4      2   0   
3   56    1       120   236    0        1      178      0      0.8      2   0   
4   57    0       120   354    0        1      163      1      0.6      2   0   

   thal  target  
0     1       1  
1     2       1  
2     2       1  
3     2       1  
4     2       1

In [75]:
print("Punto 3.1: Selección con loc")
print(df.loc[[0, 2, 4, 6, 8, 10]])
print(df.loc[:,["age","sex"]])


Punto 3.1: Selección con loc
    age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0    63    1   3       145   233    1        0      150      0      2.3   
2    41    0   1       130   204    0        0      172      0      1.4   
4    57    0   0       120   354    0        1      163      1      0.6   
6    56    0   1       140   294    0        0      153      0      1.3   
8    52    1   2       172   199    1        1      162      0      0.5   
10   54    1   0       140   239    0        1      160      0      1.2   

    slope  ca  thal  target  
0       0   0     1       1  
2       2   0     2       1  
4       2   0     2       1  
6       1   0     2       1  
8       2   0     3       1  
10      2   0     2       1  
     age  sex
0     63    1
1     37    1
2     41    0
3     56    1
4     57    0
..   ...  ...
298   57    0
299   45    1
300   68    1
301   57    1
302   57    0

[303 rows x 2 columns]


In [86]:
print("Punto 3.2: Selección con iloc:")
print(df.iloc[3:9])


Punto 3.2: Selección con iloc:
   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   
5   57    1   0       140   192    0        1      148      0      0.4      1   
6   56    0   1       140   294    0        0      153      0      1.3      1   
7   44    1   1       120   263    0        1      173      0      0.0      2   
8   52    1   2       172   199    1        1      162      0      0.5      2   

   ca  thal  target  
3   0     2       1  
4   0     2       1  
5   0     1       1  
6   0     2       1  
7   0     3       1  
8   0     3       1  


In [65]:
print("Punto 3.3: Selección de columnas por nombre:")
print(df[["age", "sex", "target"]].head())

Punto 3.3: Selección de columnas por nombre:
   age  sex  target
0   63    1       1
1   37    1       1
2   41    0       1
3   56    1       1
4   57    0       1


In [77]:
print("Punto 3.4: Selección de columnas por posición:")
print(df.iloc[:, [0, 1, -1]].head())

Punto 3.4: Selección de columnas por posición:
   age  sex  target
0   63    1       1
1   37    1       1
2   41    0       1
3   56    1       1
4   57    0       1


## Que diferencia hay entre loc e iloc?
loc permite la seleccion a traves de etiquetas y ubicacion de los elementos, en cambio iloc se basa en el indice de los eleementos por eso al seleccionar df.iloc[3:9] solo se selecciona desde el 3 hasta el 8

In [112]:
print("Punto 4.1: Ordenar por colesterol:")
print(df.sort_values(by="chol", ascending = False).head())


Punto 4.1: Ordenar por colesterol:
     age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
85    67    0   2       115   564    0        0      160      0      1.6   
28    65    0   2       140   417    1        0      157      0      0.8   
246   56    0   0       134   409    0        0      150      1      1.9   
220   63    0   0       150   407    0        0      154      0      4.0   
96    62    0   0       140   394    0        0      157      0      1.2   

     slope  ca  thal  target  
85       1   0     3       1  
28       2   1     2       1  
246      1   2     3       0  
220      1   3     3       0  
96       1   0     2       1  


In [108]:
print("Punto 4.2: Promedios por sexo:")
print(df.groupby("sex").mean())

Punto 4.2: Promedios por sexo:
           age        cp    trestbps        chol      fbs   restecg  \
sex                                                                   
0    55.677083  1.041667  133.083333  261.302083  0.12500  0.572917   
1    53.758454  0.932367  130.946860  239.289855  0.15942  0.507246   

        thalach     exang   oldpeak     slope        ca      thal    target  
sex                                                                          
0    151.125000  0.229167  0.876042  1.427083  0.552083  2.125000  0.750000  
1    148.961353  0.371981  1.115459  1.386473  0.811594  2.400966  0.449275  


In [114]:
print("Punto 4.3: Conteo por condición cardíaca:")
print(df.groupby("target").count())

Punto 4.3: Conteo por condición cardíaca:
        age  sex   cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
target                                                                         
0       138  138  138       138   138  138      138      138    138      138   
1       165  165  165       165   165  165      165      165    165      165   

        slope   ca  thal  
target                    
0         138  138   138  
1         165  165   165  


**Preguntas:**
- Qué paciente tiene el nivel de colesterol más alto según el ordenamiento?
    **El paciente identificado con el id 85 con un nivel de colesterol de 564**
- Qué diferencias observa entre los promedios de hombres (sex=1) y mujeres (sex=0)?
    **Que el promedio de mujeres enfermas es mayor al de los hombres, esto se refleja tambien en los niveles de colesteros donde las mujeres tienden a tener mas alto el colesterol que los hombres**
    
- Según el conteo por condición cardíaca, hay más pacientes con o sin enfermedad cardíaca?
  **Hay mas personas con enfermedad cardiaca, 138 sin enfermedad y 165 con enfermedad**

## Análisis Final:
1. ¿Cuál de los métodos estudiados le parece más útil para explorar datos y por qué?
   **Me parece que el método loc es muy útil, especialmente cuando se trabaja con etiquetas de índice o nombres de columnas. Permite acceder de manera clara y explícita a subconjuntos del DataFrame, lo que es muy útil en etapas de exploración donde el significado semántico de los datos importa más que su posición. Tambien el método .describe() hace muy buen resumen estadistico de la data que se esta trabajando permitiendo ver las estadisticas mas importantes de la data.**
3. ¿Explique en qué situaciones usar **loc** en lugar de **iloc**? **Se usa loc cuando queremos seleccionar filas o columnas usando etiquetas. Es muy util para ser mas expresivo en el codigo y mas claro. En cambio, como su nombre lo indica, se usa iloc cuando queremos seleccionar por posición (índices numéricos), sin importar los nombres reales.**
5. Consultar que hace el comando: **inplace=True**
   **Lo que hace es modificar el df real, la referencia como tal, por ejemplo cuando se eliminan elementos NO se mutaba el df original, se creaba uno a partir de ese, con el inplace=True si se muta la referencia original del df.**